src/Hl7.Fhir.Core/Serialization/SerializationUtil.cs

```diff
@@ -0,0 +1,485 @@
+            {
+                var doc = SerializationUtil.XDocumentFromXmlText(xmlText);
+                return RunFhirXhtmlSchemaValidation(doc);
+            }
+            catch (FormatException fe)
+            {
+                return new[] { fe.Message };
+            }
+        }
+
+        public static string[] RunFhirXhtmlSchemaValidation(XDocument doc)
+        {
+            var result = new List<string>();
+
+            if (!doc.Root.AtXhtmlDiv())
+                return new[] { $"Root element of XHTML is not a <div> from the XHTML namespace ({XmlNs.XHTML})." };
+
+            doc.Validate(_xhtmlSchemaSet.Value, (s, a) => result.Add(a.Message));
+            return result.ToArray();
+        }
+
+        private static Lazy<XmlSchemaSet> _xhtmlSchemaSet = new Lazy<XmlSchemaSet>(compileXhtmlSchema, true);
+
+        private const string XML_XSD_RESOURCENAME = "Hl7.Fhir.xhtml.xml.xsd";
+        private const string FHIRXHTML_XSD_RESOURCENAME = "Hl7.Fhir.xhtml.fhir-xhtml.xsd";
+
+        private static Lazy<string> XmlXsdData = new Lazy<string>(() => readResource(XML_XSD_RESOURCENAME));
+        private static Lazy<string> FhirXhtmlXsdData = new Lazy<string>(() => readResource(FHIRXHTML_XSD_RESOURCENAME));
+
+        private static XmlSchemaSet compileXhtmlSchema()
+        {
+            XmlSchemaSet schemas = new XmlSchemaSet();
+
+            var schema = new StringReader(XmlXsdData.Value);
+            schemas.Add(null, XmlReader.Create(schema));   // null = use schema namespace as specified in schema file
+
+            schema = new StringReader(FhirXhtmlXsdData.Value);
+            schemas.Add(null, XmlReader.Create(schema));   // null = use schema namespace as specified in schema file
+
+            schemas.Compile();
+
+            return schemas;
+        }
+
+        private static string readResource(string resourceName)
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            using (var stream = assembly.GetManifestResourceStream(resourceName))
+            {
+                if (stream == null)
+                {
+                    throw new Exception($"Resource {resourceName} not found in {assembly.FullName}.  Valid resources are: {String.Join(", ", assembly.GetManifestResourceNames())}.");
+                }
+                using (var reader = new StreamReader(stream))
+                {
+                    return reader.ReadToEnd();
+                }
+            }
+        }
+#endif
+
+
+
+#if NETSTANDARD1_1
+        private static Regex _re = new Regex("(&[a-zA-Z0-9]+;)", RegexOptions.CultureInvariant);
+#else
+        private static Regex _re = new Regex("(&[a-zA-Z0-9]+;)", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+#endif
+        private static Dictionary<string, string> _xmlReplacements;
+        private static Dictionary<string, string> getXmlReplacements()
+        {
+            if (_xmlReplacements != null)
+                return _xmlReplacements;
```

Comment:
can we set `_xmlReplacements` in a static constructor instead of lazily?

Reply:
Done